pkgs/applications/version-management/sr-ht/core.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenv, fetchgit, fetchNodeModules, buildPythonPackage
+, flask, flask_login, humanize, sqlalchemy, sqlalchemy-utils, psycopg2, markdown
+, bleach, requests, beautifulsoup4, pgpy, misaka, pygments
+, sassc, nodejs-11_x }:
+
+buildPythonPackage rec {
+  pname = "core-sr-ht";
+  version = "0.30.5";
+
+  src = fetchgit {
+    url = "https://git.sr.ht/~sircmpwn/core.sr.ht";
+    rev = version;
+    sha256 = "0ccbymmixldf0mb8dmcij2j7v0cxp1vhrfbl6az1hv5y1rxijqb6";
+  };
+
+  node_modules = fetchNodeModules {
+    src = "${src}/srht";
+    nodejs = nodejs-11_x;
+    sha256 = "0axl50swhcw8llq8z2icwr4nkr5qsw2riih0a040f9wx4xiw4p6p";
+  };
+
+  patches = [
+    ./disable-npm-install.patch
+  ];
+
+  propagatedBuildInputs = [
+    flask
+    flask_login
+    humanize
+    sqlalchemy
+    sqlalchemy-utils
+    psycopg2
+    markdown
+    bleach
+    requests
+    beautifulsoup4
+    pgpy
+    misaka
+    pygments
+
+    sassc
+    nodejs-11_x
```

Comment:
This should not be propagated.

Reply:
Every package that calls `srht` as a dependency then calls its `Makefile` which requires `sassc` and `node` to compile?

Reply:
No, because there might some package depending on e.g. sr-ht-git but not requiring them. You should put them in each package's `buildInputs`.

Reply:
Is it better to move `sassc` and `nodejs` to `buildInputs` then pass along the `buildInputs` by referencing `buildInputs = srht.buildInputs`?

Reply:
That's fine as well.